main.go

```diff
@@ -108,6 +113,17 @@ func main() {
 		setupLog.Error(err, "unable to create controller", "controller", "Gatekeeper")
 		os.Exit(1)
 	}
+	if err = (&controllers.ConstraintStatusReconciler{
```

Comment:
If the Gatekeeper operator is installed and Gatekeeper itself is not, then the CRDs needed for this controller are not there. What happens? Does it fail to start or are there error logs?

Reply:
I had the same issue in the gatekeeper-status-sync controller in the governance-policy-framework-addon, so I created a health check that causes Kubernetes to restart the pods if the installation status of Gatekeeper changes:
https://github.com/open-cluster-management-io/governance-policy-framework-addon/blob/f0e061c0ef033fb41aafe801b48753ad219c6def/controllers/gatekeepersync/health.go#L26

That way in `main.go`, you can conditionally enable this controller if Gatekeeper is installed.

Reply:
my plan was to install crds with this controller